internal/record/record.go

```diff
@@ -0,0 +1,123 @@
+package record
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"gopkg.in/yaml.v3"
+
+	"github.com/DanWlker/remind/internal/config"
+	"github.com/DanWlker/remind/internal/data"
+	i_error "github.com/DanWlker/remind/internal/error"
+)
+
+func GetFile() (string, error) {
+	dataFolder, err := data.GetFolder()
+	if err != nil {
+		return "", fmt.Errorf("helper.GetDataFolder: %w", err)
+	}
+
+	fullpath := filepath.Join(dataFolder, config.DefaultDataRecordFullFileName)
+
+	if _, err := os.Stat(fullpath); errors.Is(err, os.ErrNotExist) {
+		if _, err := os.Create(fullpath); err != nil {
+			return "", fmt.Errorf("os.Create: %w", err)
+		}
+		globalRecordEntity, err := CreateNewRecord("")
+		if err != nil {
+			return "", fmt.Errorf("CreateNewRecord: %w", err)
+		}
+
+		if err := SetFileContents([]RecordEntity{globalRecordEntity}); err != nil {
+			return "", fmt.Errorf("SetRecordFileContents: %w", err)
+		}
+	} else if err != nil {
+		return "", fmt.Errorf("os.Stat: %w", err)
+	}
+
+	return fullpath, nil
+}
+
+func GetFileContents() ([]RecordEntity, error) {
+	recordFile, err := GetFile()
+	if err != nil {
+		return nil, fmt.Errorf("GetRecordFile: %w", err)
```

Comment:
I think this was previously returning `[]RecordEntity{}`. But `nil` is the better way to express "an empty slice." It works the same way in all the contexts you care about - `len`, `append`, etc. - and unlike `[]RecordEntity{}` does not allocate an empty backing array for the [slice header](https://go.dev/blog/slices#slices-the-slice-header) to point to.